scipy/stats/tests/test_distributions.py

```diff
@@ -971,6 +971,109 @@ def test_stats(self):
         assert_allclose(result, expected)
 
 
+class TestKSTwo(object):
+    def setup_method(self):
+        np.random.seed(1234)
+
+    def test_cdf(self):
+        for n in [1, 2, 3, 10, 100, 1000]:
+            x = np.array([0, 0.5/n, 1/n, 0.5, 1-1.0/n, 1])
+            v1 = (1.0/n)**n
+            lg = scipy.special.gammaln(n+1)
+            elg = (np.exp(lg) if v1 != 0 else 0)
+            expected = np.array([0, 0, v1 * elg,
+                                 1 - 2*stats.ksone.sf(0.5, n),
+                                 max(1 - 2*v1, 0.0),
+                                 1.0])
+            vals_cdf = stats.kstwo.cdf(x, n)
+            assert_allclose(vals_cdf, expected)
+
+    def test_sf(self):
+        x = np.linspace(0, 1, 11)
+        for n in [1, 2, 3, 10, 100, 1000]:
+            x = np.array([0, 0.5/n, 1/n, 0.5, 1-1.0/n, 1])
+            v1 = (1.0/n)**n
+            lg = scipy.special.gammaln(n+1)
+            elg = (np.exp(lg) if v1 != 0 else 0)
+            expected = np.array([1.0, 1.0,
+                                 1 - v1 * elg,
+                                 2*stats.ksone.sf(0.5, n),
+                                 min(2*v1, 1.0), 0])
+            vals_cdf = stats.kstwo.sf(x, n)
+            assert_allclose(vals_cdf, expected)
+
+    def test_cdf_sqrtn(self):
+        x = np.linspace(0, 2, 11)[1:]
+        # ns = [1, 2, 3, 10, 100, 1000]
+        ns = [50, 100, 200, 400, 1000, 2000]
+        for _x in x:
+            xn = _x / np.sqrt(ns)
+            probs = stats.kstwo.cdf(xn, ns)
+            diffs = np.diff(probs)
+            cond = (diffs >= 1e-8)
+            assert len(diffs[cond]) == 0
```

Comment:
where does this result come from?

Reply:
For fixed `a`, as `n->infinity`, `kstwo.cdf(a/sqrt(n), n) -> kstwo.cdf(a, n)`, and it does so monotonically increasing.  This check just ensures that the sequence is increasing, up to some small floating point jitter.  I've aded some comments.